src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -58,12 +61,17 @@ internal class SourceInfo : ITaintedDataInfo, IEquatable<SourceInfo>
         /// <summary>
         /// Methods that generate tainted data and whose arguments don't need extra value content analysis.
         /// </summary>
-        public ImmutableDictionary<string, IsInvocationTaintedWithPointsToAnalysis> TaintedMethodsNeedPointsToAnalysis { get; }
+        public ImmutableDictionary<MethodMapper, ImmutableDictionary<PointsToCheck, ImmutableHashSet<string>>> TaintedMethodsNeedsPointsToAnalysis { get; }
 
         /// <summary>
         /// Methods that generate tainted data and whose arguments need extra value content analysis and points to analysis.
         /// </summary>
-        public ImmutableDictionary<string, IsInvocationTaintedWithValueContentAnalysis> TaintedMethodsNeedsValueContentAnalysis { get; }
+        public ImmutableDictionary<MethodMapper, ImmutableDictionary<ValueContentCheck, ImmutableHashSet<string>>> TaintedMethodsNeedsValueContentAnalysis { get; }
+
+        /// <summary>
+        /// Methods that could taint another argument when one of its argument is tainted.
+        /// </summary>
+        public ImmutableDictionary<MethodMapper, ImmutableHashSet<(string, string)>> PasserMethods { get; }
```

Comment:
If we're just going to access these by doing a linear search, then maybe we should just have arrays of tuples, e.g.:
```csharp
ImmutableArrays<(MethodMapper MethodMapper, ImmutableDictionary<PointsToCheck, ImmutableHashSet<string>> PointsToChecks)>
```

Reply:
Why HashSets instead of Arrays?

Reply:
In case that people set up two same SourceInfos.